exec/plugin-commands-script-runners/src/buildCommandNotFoundHint.ts

```diff
@@ -9,11 +9,25 @@ export function getNearest (name: string, list?: readonly string[]) {
   })
 }
 
-export async function getNearestProgram (programName: string) {
+export async function getNearestProgram (opts: {
+  programName: string,
+  dir: string,
+  workspaceDir: string | undefined,
+}) {
   try {
-    const binDir = path.join(process.cwd(), 'node_modules', '.bin')
-    const programList = await readdir(binDir)
-    return getNearest(programName, programList)
+    const { programName, dir, workspaceDir } = opts
+    const binDir = path.join(dir, 'node_modules', '.bin')
+    const programListPromise = readdir(binDir) // await is omitted to take advantage of concurrency
+    let programList!: string[]
+    if (workspaceDir && workspaceDir !== dir) {
+      const workspaceBinDir = path.join(workspaceDir, 'node_modules', '.bin')
+      const programListExtension = await readdir(workspaceBinDir)
```

Comment:
I think it is OK if you just use readdir synchronously here and make the function synchronous. We don't care about the performance here so much.